src/Interpreters/HashJoin.cpp

```diff
@@ -1224,19 +1235,18 @@ NO_INLINE IColumn::Filter joinRightColumns(
                 auto & mapped = find_result.getMapped();
                 if constexpr (jf.is_asof_join)
                 {
-                    TypeIndex asof_type = added_columns.asofType();
-                    ASOF::Inequality asof_inequality = added_columns.asofInequality();
                     const IColumn & left_asof_key = added_columns.leftAsofKey();
 
-                    if (const RowRef * found = mapped.findAsof(asof_type, asof_inequality, left_asof_key, i))
+                    auto [block, row_num] = mapped->findAsof(left_asof_key, i);
```

Comment:
This looks much easy to understand, than before, we embedded types and asof type into mapped object and just call `findAsof` method.

Reply:
Yep. It's simpler and faster.